---
- name: initial start of ethereum python client, connection to bootstrapping node
  docker: 
      image:    "{{ docker_image_py }}"
      name:     "{{ container_id }}"     
      expose:   30303
      ports:    30303:30303
      
      # parameters given to client 
      command: --logging :debug --log_json 1 --remote {{ bootstrap_ip }} --port 30303 --mining {{ mining_cpu_percentage }} --peers {{ req_num_peers }}

- name: Log docker output to elarch
  # fire and forget, don't wait to finish, run for one day
  async: 86400
  poll: 0
  # again, ansible docker module does not support logs
  # pyethereum wrongly logs to stderr
  shell: docker logs --follow=true {{ container_id }} 2>&1 | {{ repo_path }}/logging/teees.py {{ elarch_ip }}
  register: log_runner

- name: verify previous task
  async_status: jid={{ log_runner.ansible_job_id }}
